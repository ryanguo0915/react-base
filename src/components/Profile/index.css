.profile__title {
  font-weight: bold;
  text-align: center;
}
.profile__info {
  display: flex;
}
.profile__table {
  max-width: 300px;
}
.profile__table td:first-child {
  width: 100px;
  font-weight: bold;
}

.profile__table--bio td:first-child {
  vertical-align: baseline;
}
