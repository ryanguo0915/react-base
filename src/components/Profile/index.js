import React  from "react";
import "./index.css";

export default function Profile({profile}) {
    const {avatarImage, bio,email, firstName, lastName, phone} = profile;
  return (
    <div className="profile__wrapper">
      <h3 className="profile__title">
        Profile
      </h3>
      <div className="profile__info">
        <img src={avatarImage} height={200} width={200} alt="avatar"/>
        <table className="profile__table">
            <tbody>
                <tr>
                    <td>First Name</td>
                    <td>{firstName}</td>
                </tr>
                <tr>
                    <td>Last Name</td>
                    <td>{lastName}</td>
                </tr>
                <tr>
                    <td>Phone</td>
                    <td>{phone}</td>
                </tr>
                <tr>
                    <td>Email</td>
                    <td>{email}</td>
                </tr>            
                <tr className="profile__table--bio">
                    <td>Bio</td>
                    <td>{bio}</td>
                </tr>            
            </tbody>
        </table>
      </div>
    </div>
  );
}