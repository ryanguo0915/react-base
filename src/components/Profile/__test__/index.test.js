import React from 'react';
import { shallow } from 'enzyme';
import Profile from '../index';
describe("NarBar", () => {
  it("should render my component", () => {
    const props = { 
      profile: {avatarImage:'imageMock', bio: 'bioMock', email: 'emailMock', firstName: 'First Name', lastName: 'Last Name', phone: 'Phone'} 
    };
    const wrapper = shallow(<Profile {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});