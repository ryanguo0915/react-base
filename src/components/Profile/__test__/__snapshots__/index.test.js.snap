// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NarBar should render my component 1`] = `
Array [
  <div
    className="profile__wrapper"
  >
    <h3
      className="profile__title"
    >
      Profile
    </h3>
    <div
      className="profile__info"
    >
      <img
        alt="avatar"
        height={200}
        src="imageMock"
        width={200}
      />
      <table
        className="profile__table"
      >
        <tbody>
          <tr>
            <td>
              First Name
            </td>
            <td>
              First Name
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Last Name
            </td>
          </tr>
          <tr>
            <td>
              Phone
            </td>
            <td>
              Phone
            </td>
          </tr>
          <tr>
            <td>
              Email
            </td>
            <td>
              emailMock
            </td>
          </tr>
          <tr
            className="profile__table--bio"
          >
            <td>
              Bio
            </td>
            <td>
              bioMock
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
